config/common.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"path/filepath"
+
+	"github.com/prometheus/common/config"
+)
+
+// JoinDir returns path relative to dir.
+// If path is empty or absolute, it is return unchanged.
+func JoinDir(dir, path string) string {
```

Comment:
I struggled with naming this... Technically, it doesn't ensure an absolute path because dir is allowed to be relative... and I didn't want to change the existing behavior. It joins dir to path, if path is join-able.